controllers/operandconfig.go

```diff
@@ -241,29 +250,30 @@ func (r *CommonServiceReconciler) updateOperandConfig(newConfigs []interface{})
 			}
 			if rules != nil && rules.(map[string]interface{})["spec"] != nil && rules.(map[string]interface{})["spec"].(map[string]interface{})[cr] != nil {
 				ruleForCR := rules.(map[string]interface{})["spec"].(map[string]interface{})[cr].(map[string]interface{})
-				opService.(map[string]interface{})["spec"].(map[string]interface{})[cr] = mergeCRsIntoOperandConfig(spec.(map[string]interface{}), newConfigForCR, ruleForCR, overwrite)
+				opService.(map[string]interface{})["spec"].(map[string]interface{})[cr], isChangedWithRules = mergeCRsIntoOperandConfig(spec.(map[string]interface{}), newConfigForCR, ruleForCR, overwrite)
 			} else {
 				if overwrite {
-					opService.(map[string]interface{})["spec"].(map[string]interface{})[cr] = mergeCRsIntoOperandConfigWithDefaultRules(spec.(map[string]interface{}), newConfigForCR)
+					opService.(map[string]interface{})["spec"].(map[string]interface{})[cr], isChangedWithDefultRules = mergeCRsIntoOperandConfigWithDefaultRules(spec.(map[string]interface{}), newConfigForCR)
 				}
 			}
+			isChanged = isChanged || isChangedWithRules || isChangedWithDefultRules
 		}
 	}
 
 	// Checking all the common service CRs to get the minimal size
-	opconServices, err = r.getMinimalSizes(opconServices, ruleSlice)
+	opconServices, _, err = r.getMinimalSizes(opconServices, ruleSlice)
 	if err != nil {
-		return err
+		return false, err
 	}
 
 	opcon.Object["spec"].(map[string]interface{})["services"] = opconServices
```

Comment:
@Daniel-Fan Could we just make a comparison here? if the services don't change then report an event. Then we don't need to pass the ischange button up.

Reply:
The simple DeepCopy and DeepEqual did not work in this case.

Will try a loop to iterate the `opconServices` for the comparison.